Zend/zend_vm_def.h

```diff
@@ -6340,6 +6367,97 @@ ZEND_VM_HANDLER(160, ZEND_YIELD, CONST|TMP|VAR|CV|UNUSED, CONST|TMP|VAR|CV|UNUSE
 	ZEND_VM_RETURN();
 }
 
+ZEND_VM_HANDLER(170, ZEND_YIELD_FROM, CONST|TMP|VAR|CV, ANY)
+{
+	USE_OPLINE
+
+	/* The generator object is stored in EX(return_value) */
+	zend_generator *generator = (zend_generator *) EX(return_value);
+
+	zval *val;
+	zend_free_op free_op1;
+
+	SAVE_OPLINE();
+	val = GET_OP1_ZVAL_PTR_DEREF(BP_VAR_R);
+	if (Z_TYPE_P(val) == IS_ARRAY) {
+		ZVAL_COPY_VALUE(&generator->values, val);
+		if (OP1_TYPE != IS_TMP_VAR && Z_OPT_REFCOUNTED_P(val)) {
+			Z_ADDREF_P(val);
+		}
+		Z_FE_POS(generator->values) = 0;
+
+		FREE_OP1_IF_VAR();
+	} else if (OP1_TYPE != IS_CONST && Z_TYPE_P(val) == IS_OBJECT && Z_OBJCE_P(val)->get_iterator) {
+		zend_class_entry *ce = Z_OBJCE_P(val);
+		if (ce == zend_ce_generator) {
+			zend_generator *new_gen = (zend_generator *) Z_OBJ_P(val);
+
+			if (OP1_TYPE != IS_TMP_VAR) {
+				Z_ADDREF_P(val);
+			}
+			FREE_OP1_IF_VAR();
+
+			if (Z_ISUNDEF(new_gen->retval)) {
+				zend_generator_yield_from(generator, new_gen);
+			} else if (new_gen->execute_data == NULL) {
+				// TODO: Should be an engine exception
+				zend_error(E_RECOVERABLE_ERROR, "Generator passed to yield from was aborted without proper return and is unable to continue");
+
+				ZEND_VM_NEXT_OPCODE();
+			} else {
+				if (RETURN_VALUE_USED(opline)) {
+					ZVAL_COPY(EX_VAR(opline->result.var), &new_gen->retval);
+				}
+
+				ZEND_VM_NEXT_OPCODE();
+			}
```

Comment:
By exception check I mean a `CHECK_EXCEPTION()` call before `ZEND_VM_NEXT_OPCODE()`, no need to test `EG(exception)` or something like that ;)